### DISCUSSION TITLE
AIRFLOW ULTIMATE NIGHTMARE

### DISCUSSION THREAD

**Discussion by torracc** [apache/airflow#36172] [Q&A]
Hello everyone from the community, how are you?

For me, there is no more hope, so I come to you masters of opensource wisdom, to help me with my newest chapter of terror and mental disorder.

I'm trying to make Airflow available on Kubernetes.

The data engineering team created a Hello World DAG for me (Infrastructure) to test.

I'm trying the approach via kubernetesexecutor and unfortunately it's not working.

The DAG goes into running mode and does not come out of there.

I'm using Manifests .yaml instead of Helm Chart.

Postgres ok
Redis ok
PV and PVCs for postgres, redis and airflow ok.

My questions are the following:

1 - How does Airflow communicate with the cluster to upload worker pods? (Where and how to configure.)

2 - I'm trying to run the DAGs inside the worker image.
Which image should I use for the worker? Does it receive all packages from the main image?

3 - I'm using pod_template_file in airflow.cfg. Is there any hidden secret for airflow to use the file? (I followed the guidance in the documentation to create it.)

4 - My cluster is not exclusive to airflow and there are other applications running on it.

What information should I provide so that I can receive some help from the community?

Thank you very much in advance.

**Comment by potiuk** [comment]
Use helm chart.  If you do not want to use it use `helm install --dry-run` and `helm-diff` plugin. This will produce the resources that you can look at and create your own resources based on that. This is the fastest way to learn it.

**Reply by torracc** [reply]
Thanks for the advice. I'll try this way out and let you know asap I get the results. :)
